### PR TITLE
ARTEMIS-1924 small tweaks on HeartBeat test

### PR DESCRIPTION
**Pull request by clebertsuconic**

### CODE REVIEW

**Comment by clebertsuconic** [comment]
@gemmellr / @mtaylor can you review this?

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/AmqpNoHearbeatsTest.java [hunk 1]
```diff
@@ -137,18 +139,21 @@ public void inspectOpenedResource(Connection connection) {
    }
 
    public static void main(String[] arg) {
-      if (arg.length > 0 && arg[0].equals("testConnectionReset")) {
+      if (arg.length > 0 && arg[0].startsWith("testCloseConsumerOnConnectionReset")) {
```
Comment (gemmellr) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/AmqpNoHearbeatsTest.java]: How about using the argument as the queue name? That way you dont need a constant, and there are side benefits such as test isolation and if things to wonky you can read the logs and immediately know what test the output came from etc?

Comment (clebertsuconic) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/AmqpNoHearbeatsTest.java]: I was going to do that, but then the argument was: testCloseConsumerOnConnectionReset[useOverride=true]() 

The test is using parameters.. which I didn't like as the name of the queue... 

I would prefer leaving it as is...

**Comment by gemmellr** [comment]
Looks good

**Comment by clebertsuconic** [comment]
@gemmellr thanks.. I will merge it then

**Comment by gemmellr** [comment]
(I still don't think it should be in that class though per #2149 comments...but the test itself looks good now)

**Comment by clebertsuconic** [comment]
@gemmellr that I disagree with.. this is about no heart beat.. and disconnecting the client.

**Comment by gemmellr** [comment]
To me it seemed more about cleaning up the client regardless of idle-timeout, and that it should do exactly the same if idle-timeout handling is enabled or if it wasn't using AMQP at all.

**Comment by clebertsuconic** [comment]
@gemmellr the issue i was dealing with was a .NET client in windows not cleaning up after a disconnect, after I created the feature. So.. this is definitely related. We can create more tests outside of this context.. but this test is definitely needed IMO.

**Comment by gemmellr** [comment]
Yes, after a connection-reset-by-peer error which meant the broker knew the connection had gone, which it knew regardless of whether idle-timeout handling was enabled or not, but failed to clear it up at the time. I think we may need to agree to disagree here :)

**Comment by clebertsuconic** [comment]
with heart beats.. you won't this test.. as ping/pongs will disconnect the client.
Disabling heart beats and still solving issues it's all that this fix was about and the reason I created these tests.

**Comment by clebertsuconic** [comment]
It's pointless to argue anyway.. it's just a test and it's working as expected... I need to move to a different task ;)

**Comment by gemmellr** [comment]
> with heart beats.. you won't this test.. as ping/pongs will disconnect the client.

The idle-timeout stuff defaults to killing after 60 seconds, which the test doesnt even allow for running long enough to occur, so disabling heartbeats should make no difference to the test, which is why I dont think it should be in this class. But I agree, lets move on.

**Pull request merged**